packages/design-system/src/components/Button/variations/ButtonSecondary.tsx

```diff
@@ -1,14 +1,25 @@
 import React, { forwardRef, Ref } from 'react';
-import ButtonPrimitive, { BaseButtonProps } from '../Primitive/ButtonPrimitive';
+import ButtonPrimitive, { AvailableSizes, BaseButtonProps } from '../Primitive/ButtonPrimitive';
 
 import styles from './ButtonSecondary.module.scss';
+import { ButtonDestructivePropsType } from './ButtonDestructive';
 
-export type ButtonSecondaryPropsType = Omit<BaseButtonProps, 'className'>;
+export type ButtonSecondaryPropsType<S extends AvailableSizes> = Omit<
+	BaseButtonProps<S>,
+	'className' | 'size'
+> & {
+	size?: S;
+};
 
-const ButtonSecondary = forwardRef(
-	(props: ButtonSecondaryPropsType, ref: Ref<HTMLButtonElement>) => {
-		return <ButtonPrimitive {...props} ref={ref} className={styles.secondary} />;
-	},
-);
+function Secondary<S extends AvailableSizes>(
+	props: ButtonSecondaryPropsType<S>,
+	ref: Ref<HTMLButtonElement>,
+) {
+	return <ButtonPrimitive {...props} ref={ref} className={styles.secondary} />;
```

Comment:
Don't we need to 
`const { size = 'M', ...rest } = props;` on this button too ?